src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -470,7 +476,8 @@ QgsOgrProvider::QgsOgrProvider( QString const &uri, const ProviderOptions &optio
       << QgsVectorDataProvider::NativeType( tr( "Whole number (integer)" ), QStringLiteral( "integer" ), QVariant::Int, 0, nMaxIntLen )
       << QgsVectorDataProvider::NativeType( tr( "Whole number (integer 64 bit)" ), QStringLiteral( "integer64" ), QVariant::LongLong, 0, nMaxInt64Len )
       << QgsVectorDataProvider::NativeType( tr( "Decimal number (real)" ), QStringLiteral( "double" ), QVariant::Double, 0, nMaxDoubleLen, 0, nMaxDoublePrec )
-      << QgsVectorDataProvider::NativeType( tr( "Text (string)" ), QStringLiteral( "string" ), QVariant::String, 0, 65535 );
+      << QgsVectorDataProvider::NativeType( tr( "Text (string)" ), QStringLiteral( "string" ), QVariant::String, 0, 65535 )
+      << QgsVectorDataProvider::NativeType( tr( "Map (JSON)" ), QStringLiteral( "JSON" ), QVariant::Map, 0, 0, 0, 0, QVariant::String );
```

Comment:
Should this be "JSON (string)"? JSON also supports arrays and the internal representation (in gpkg) is a string with a type hint.

Reply:
You mean just the label? Still working with `QVariant::Map` right?

Reply:
I was talking about the label, @nyalldawson's comment is also valid (in the future we could think about another way of hinting type promotion to qgsogrprovider, saved in the qgis project, but that's a different story).

Reply:
To make it coherent to the Postgres types json and jsonb (at the moment "Map (json)" and "Map (jsonb)") I would suggest to use the following labels:
- Postgres json: "JSON (json)"
- Postgres jsonb: "JSON (jsonb)"
- GPKG string (json): "JSON (string)"
Because in Postgres it can be a list as well. And in future even value relation.
What do you think?

Reply:
:+1: I like